src/java/org/apache/cassandra/auth/IInternodeAuthenticator.java

```diff
@@ -35,10 +36,51 @@
      */
     boolean authenticate(InetAddress remoteAddress, int remotePort);
 
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(Certificate[] certificates)
+    {
+        return false;
+    }
+
+
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param remoteAddress ip address of the connecting node.
+     * @param remorePort port of the connecting node.
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(InetAddress remoteAddress, int remorePort, Certificate[] certificates)
+    {
+        if (certificates != null)
+        {
+            return authenticate(certificates);
+        }
```

Comment:
I think this can be replaced with a comment saying something like `Default implementation calls the deprecated authenticate by IP and port method`.